### PR TITLE
Generate transpiled sources before publishing to npm

### PR DESCRIPTION
**Pull request by ahutchings**
The browserify support that was added recently was broken because it did not include babelify or the babel plugins in the "dependencies" field. I refactored the build pipeline to do the transpile at publish-time rather than when the module is consumed by a bundler so that you will not need any extra configuration to support browserify. I added the jsnext:main field to package.json which should make this module consumable by rollup, but I did not test that part as I do not use rollup.

* Add a prepublish script that transpiles src/ into lib/
* Add lib and src to package.json "files" field so that they are uploaded to the npm registry
* Point the main field to lib/index.js for non-ES2015 consumers (node, webpack, browserify)
* Add jsnext:main field that points to the ES2015 version of the source for rollup and other ES2015-consuming bundlers (https://github.com/rollup/rollup/wiki/jsnext:main)
* Remove browserify configuration that is no longer needed since the transpile happens at publish time

NOTE: I had to change the shape of the exports to `import { Sortable } from 'react-sortable';` so that both ES2015 and non-ES2015 consumers get the same API. This will need to be a major version bump when publishing a new version to npm.

### CODE REVIEW

**Comment by Dharmoslap** [comment]
Thanks for PR. I would prefer to have there webpack, or anything for watching files and rebuilding a bundle on save. I also think that we should have there compiled JS bundle as a part of repository, to have a working demo running without setup.

File: webpack.config.js [hunk 1]
```diff
@@ -1,24 +0,0 @@
...
-      {
-        test: /\.js?$/,
-        loader: 'babel-loader'
-      }
-    ]
-  },
-  plugins: [
-    new webpack.optimize.UglifyJsPlugin()
-  ]
-};
```
Comment (Dharmoslap) [inline webpack.config.js]: Why to remove webpack?

Comment (ahutchings) [inline webpack.config.js]: Babel-cli is now responsible for building the files, so webpack was not needed in the root of the repo.

**Comment by ahutchings** [comment]
Babel-cli has a watch mode - I'll enable that unless you want to use webpack. I'll also remove `lib` from the .gitignore so that the compiled files are added to the repo.

**Comment by Dharmoslap** [comment]
Alright, please enable watch mode 👍

**Comment by Dharmoslap** [comment]
Please create a new PR

**Pull request merged**